BiometricLocker/BiometricLocker.swift

```diff
@@ -15,51 +15,80 @@ public protocol AuthenticationDelegate: class {
+    }
+
+    /// Defines the UserDefaults key we'll use to store the date.
+    private enum Key: String {
+        case applicationDidEnterBackgroundDate = "com.bakkenbaeck.BiometricLocker.applicationDidEnterBackgroundDate"
     }
 
-    open weak var delegate: AuthenticationDelegate?
+    public weak var delegate: AuthenticationDelegate?
+
+    /**
+     Define which policy we have for the device owner to be authenticated using a biometric method (Touch ID or Face ID).
+     Defaults to `deviceOwnerAuthenticationWithBiometrics`.
 
-    open var policy: LAPolicy = .deviceOwnerAuthenticationWithBiometrics
+     See [LocalAuthentication](apple-reference-documentation://cslocalauthentication) for a more detailed explanation.
+     */
+    public var policy: LAPolicy = .deviceOwnerAuthenticationWithBiometrics
 
-    open var localizedReason: String = ""
+    /// The app-provided reason for requesting authentication, which displays in the authentication dialog presented to the user.
+    public var localizedReason: String = ""
 
-    open var authenticationContext: LAContext {
+    private var authenticationContext: LAContext {
         // Prevents re-using an `LAContext`, once it can no longer evaluate our policy.
         // Just reusing the `LAContext` can cause it to call the success completion block
         // without the user having to enter their biometric ID.
         if !self._authenticationContext.canEvaluatePolicy(self.policy, error: nil) {
-            self._authenticationContext = LAContext()
+            self._authenticationContext = self.newContext()
         }
 
         return self._authenticationContext
     }
 
-    // We do this to prevent a bug where quickly re-using a LAcontext
+    /// Should be set to `self.newContext()`, to ensure we're using the correct configurations.
     private var _authenticationContext = LAContext()
 
     private var defaults = UserDefaults.standard
 
-    private lazy var negativeFeedbackGenerator: UIImpactFeedbackGenerator = {
-        UIImpactFeedbackGenerator(style: .heavy)
-    }()
-
-    private lazy var positiveFeedbackGenerator: UIImpactFeedbackGenerator = {
-        UIImpactFeedbackGenerator(style: .light)
-    }()
+    private let negativeFeedbackGenerator = UIImpactFeedbackGenerator(style: .heavy)
+    private let positiveFeedbackGenerator = UIImpactFeedbackGenerator(style: .light)
 
+    /// The duration for which the biometric authentication reuse is allowable.
+    ///
+    /// If the device was successfully authenticated using biometrics within the specified time interval,
+    /// then authentication for the receiver succeeds automatically, without prompting the user again.
+    ///
+    /// - Important: Values are only valid between 0 and `LATouchIDAuthenticationMaximumAllowableReuseDuration` (checked on iOS 11.2 to be 5 minutes). If it's more, it will be reveted to `LATouchIDAuthenticationMaximumAllowableReuseDuration`, and if it's negative, to 0.
```

Comment:
nit: `reveted` ->`reverted`

Reply:
also, is the behavior mentioned here enforced by the context? it's definitely not enforced in the setter.

Reply:
It is :+1: .